This patch fixes the deprecated NumPy usage of np.int
Author: Stamen Miroslavov <stamen.miroslavov@bsc.es>
Barcelona Supercomputing Center 2025
diff -ruNBbw a/nctime/utils/time.py b/nctime/utils/time.py
--- a/nctime/utils/time.py	2025-08-27 16:28:59.269797627 +0200
+++ b/nctime/utils/time.py	2025-08-27 16:35:53.078650166 +0200
@@ -161,7 +161,7 @@
             cdftime = utime(units_as_days, calendar=calendar)
             years_axis_as_days = cdftime.date2num(years_axis)
             # Index of each years
-            yind = np.vectorize(np.int)(np.floor(num_axis_mod))
+            yind = np.vectorize(np.int64)(np.floor(num_axis_mod))
             # Rebuilt num_axis as 'days since' adding the number of days since referenced time
             # with an half-increment (num_axis_mod - yind) = 0 or 0.5
             num_axis_mod_days = (years_axis_as_days[yind - int(min_years)] +
@@ -182,7 +182,7 @@
             cdftime = utime(units_as_days, calendar=calendar)
             months_axis_as_days = cdftime.date2num(months_axis)
             # Index of each months
-            mind = np.vectorize(np.int)(np.floor(num_axis_mod))
+            mind = np.vectorize(np.int64)(np.floor(num_axis_mod))
             # Rebuilt num_axis as 'days since' adding the number of days since referenced time
             # with an half-increment (num_axis_mod - mind) = 0 or 0.5
             num_axis_mod_days = (months_axis_as_days[mind - int(min_months)] +
